test/integration/targets/mongodb_replicaset/tasks/mongod_teardown.yml

```diff
@@ -0,0 +1,25 @@
+- name: Kill all mongod processes
+  command: pkill mongod
+  ignore_errors: true
+
+- name: Ensure mongod instances have drained
```

Comment:
Is there something that can be done to the mogod configuration to speed this up? The test spends ~6 minutes on the various drain tasks. Most of them either finish immediately or take about one minute. These are by far the slowest parts of the test.

Reply:
I've attempted to use a different storage engine to no avail. The mmapv1 engine should be faster in this use case since it's not fully transactional and you can disable the journal. All my tests proved slightly slower...

> mmapv2
> 
> real	11m34.460s
> user	0m7.152s
> sys	0m2.700s
> 
> WiredTiger
> 
> real	11m29.303s
> user	0m7.584s
> sys	0m2.922s
> 

I've left this in place with the mongod_storage_engine_opts variable but gone back to wiredTiger. It's probably not a good idea any since mmapv2 is depreciated in 4.0 and they will likely removed it in a coming version.

Shutdown in WiredTiger is known to be slow. We might see improvements in a coming version. See https://jira.mongodb.org/browse/WT-4190

Also tried a brutal SIGKILL but this break things further along the line. Not recommended anyway.

I also attempted a forced shutdown via the following method...

```
- name: Force shutdown using mongo client no auth
  shell: "mongo admin --eval 'db.adminCommand({shutdown : 1, force : true})' --port {{ item }}"
  with_items:
    - 3001
    - 3002
    - 3003
  ignore_errors: True
  when: mongod_auth == False

- name: Force shutdown using mongo client with auth
  shell: "mongo admin --eval 'db.adminCommand({shutdown : 1, force : true})' --port {{ item }} -u {{ mongodb_admin_user }} -p {{ mongodb_admin_password }}"
  with_items:
    - 3001
    - 3002
    - 3003
  ignore_errors: True
  when: mongod_auth == True
```

This has no effect either so I have removed this. I'm out of ideas.

Reply:
What problems did the SIGKILL result in? That seems like the fastest way to clean things up  between stages of the tests if there's no graceful way to speed up the draining.

Reply:
Passing in SIGKILL consistently resulted in the following failed task...

> TASK [mongodb_replicaset : Validate replicaset previously created] *************
> fatal: [testhost]: FAILED! => {"changed": false, "msg": "Unable to query replica_set info: not authorized on local to execute command { find: \"system.replset\", filter: {}, $db: \"local\", $readPreference: { mode: \"primaryPreferred\" } }"}

This must screw with the data somehow meaning I can't authenticate with the user previously created. The doc do explicatly state not to do this anwaz...

> Never use kill -9 (i.e. SIGKILL) to terminate a mongod instance.

https://docs.mongodb.com/manual/tutorial/manage-mongodb-processes/

Bizarrely even that resulted in long drain times...

> TASK [mongodb_replicaset : Ensure mongod instances have drained] ***************
> ok: [testhost] => (item=3001) => {"changed": false, "elapsed": 0, "item": 3001, "match_groupdict": {}, "match_groups": [], "path": null, "port": 3001, "search_regex": null, "state": "drained"}
> ok: [testhost] => (item=3002) => {"changed": false, "elapsed": 65, "item": 3002, "match_groupdict": {}, "match_groups": [], "path": null, "port": 3002, "search_regex": null, "state": "drained"}
> ok: [testhost] => (item=3003) => {"changed": false, "elapsed": 0, "item": 3003, "match_groupdict": {}, "match_groups": [], "path": null, "port": 3003, "search_regex": null, "state": "drained"}

Which in theory should not happen and I'm unsure why it behaves like this.

So given that it's not advised, breaks things, and doesn't work as we expect anyway, I think it's a no-goer.